shared/whats-new/icon/index.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import {IconStyle} from '../../common-adapters/icon'
+import {keybaseFM} from '../../constants/whats-new'
+import Popup from '../popup.desktop'
+
+type Props = {
+  color?: string
+  badgeColor?: string
+  style?: IconStyle
+  className?: string
+  newRelease: boolean
+  onClick?: () => void
+}
+
+type PopupProps = Props & {
+  // Desktop only
+  attachToRef: React.RefObject<Kb.Box2>
+}
+
+const realCSS = `
+  .rainbowGradient {
+    -webkit-background-clip: text !important;
+  }
+`
+// Forward the ref of the icon so we can attach the FloatingBox on desktop to this component
+const Icon = (props: Props) => {
+  const badgeSize = props.badgeColor ? 8 : 12
+  const badgeSizeInner = badgeSize - 4
+
+  return props.newRelease ? (
+    Styles.isMobile ? (
+      <Kb.Icon
+        type="iconfont-radio"
+        onClick={props.onClick}
+        color={Styles.globalColors.blue}
+        style={Styles.collapseStyles([{marginRight: Styles.globalMargins.small}, props.style])}
+      />
+    ) : (
+      <>
+        <Kb.DesktopStyle style={realCSS} />
+        <Kb.Icon
+          type="iconfont-radio"
+          style={styles.rainbowColor}
+          className="rainbowGradient"
+          onClick={props.onClick}
+        />
+        <Kb.Badge
+          border={true}
+          leftRightPadding={0}
+          height={badgeSize}
+          containerStyle={Styles.collapseStyles([
+            styles.badgeContainerStyle,
+            props.badgeColor
```

Comment:
putting all the style object literals below into `styles` sounds good, might help with readability here